docs/apis/_files/locallib.mdx

```diff
@@ -0,0 +1,11 @@
+:::caution Legacy feature
```

Comment:
I love to see it has been marked as a legacy (I think that's a good idea). While reviewing the whole patch, I was wondering why the Common files (and also the xxx Plugins) are versioned. Wouldn't be better to have them unversioned and then add some badge to mark things version-dependant? 
I guess I'm missing some scenario where will be useful to have this information versioned (that's why I'm asking about it).

Reply:
I feel that this stuff is specific to Moodle development as a whole and probabyls houdl be versioned.

Right now the docs are split into:

- docs - stuff related to Moodle development which is for a version of Moodle
- general - stuff which does not need a version

However, there's also an additional divide:

- docs: Only contains development stuff
- general: Stuff which is meta to development - like Process, tooling, etc.

Ultimately there is no _harm_ to having docs in the versioned wiki where it's about development, but if we move it into general then it does becomes slightlyt harder to link between things. For example, the page tags are within a specific instance of the docs (docs, and general are different instances of the the docusaurus-plugin-docs plugin).

So tl;dr: My feeling is that these _should_ be documented. We can add things like `<Since>` tags, and legacy, deprecated, etc. and over time those will be removed when they're removed from Moodle.